Sources/HTTPTypes/HTTPFields.swift

```diff
@@ -99,7 +109,40 @@ public struct HTTPFields: Sendable, Hashable {
         }
     }
 
-    private var _storage = _Storage()
+#if canImport(Synchronization)
+    @available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+    private final class _StorageWithMutex: _Storage, @unchecked Sendable {
+        let mutex = Mutex<Void>(())
+
+        override func withLock<Result>(_ body: () throws -> Result) rethrows -> Result {
```

Comment:
Does this actually compile without warnings under strict concurrency? The closure that `Mutex` takes is `sending` whereas ours is not. This should produce a warning/error.

Reply:
Needing dynamic dispatch is also a little gnarly.

Reply:
I tried `.enableExperimentalFeature("StrictConcurrency")` and didn't get any errors, but added `sending` in 948fe6c4885ef0484237729c28d8df620da83ae4 to be safe.

Reply:
I think dynamic dispatch is still better than an extra allocation for the lock?

Reply:
Had to revert 948fe6c4885ef0484237729c28d8df620da83ae4 due to it breaking Swift 5.8/5.9